File: src/main/java/my/study/repository/design/pattern/chain/java/ConcreteHandlerB.java
package my.study.repository.design.pattern.chain.java;

/**
 * @author djh on  2018/9/16 20:23
 * @E-Mail 1544579459@qq.com
 */
class ConcreteHandlerB extends AbstractHandler {


    @Override
    void handlerRequest(int money) {
        if (money >= 30 && money < 50) {
            System.out.println("可以买宝马5系,买车成功!");
        } else {
            mNextHandler.handlerRequest(money);
        }
    }
}

File: src/main/java/my/study/repository/design/pattern/chain/kotlin/ConcreteHandlerB.kt
package my.study.repository.design.pattern.chain.kotlin

/**
 * @author djh on  2018/10/4 18:25
 * @E-Mail 1544579459@qq.com
 */
class ConcreteHandlerB : AbstractHandler() {
    override fun handlerRequest(money: Int) {
        if (money in 30..49) {
            println("${money}万,可以买宝马5系,买车成功!")
        } else {
            nextHandler.handlerRequest(money)
        }
    }
}